thunder/core/transforms.py

```diff
@@ -2373,6 +2373,9 @@ def zeta_backward(x, y, g):
     # Therefore, we compute only the derivative wrt the second argument
     gy = g * -x * prims.zeta(x + 1.0, y)
 
+    # if x is CUDA tensor and y is CPU scalar tensor, gy should be on CPU
+    if gy is not None and gy.device.type != y.device.type:
```

Comment:
fyi @IvanYashchuk -- an interesting special case, @kiya00

Reply:
Usually, the gradient tensors mirror their tensor properties. Seems correct that `gy` is moved to `y.device`.